### ISSUE TITLE
[ROKS] Cluster role system:node needs to add more permissions for volumeattachment

### ISSUE THREAD

**Issue by akgunjal** [openshift/ocs-operator#453]
The cluster role **system:node** needs to add more permissions for **volumeattachment**

This needs to be part of OCS deployment where RBAC is defined.

**Comment by travisn** [comment]
More details on the failure and workaround are needed, thanks

**Comment by akgunjal** [comment]
@travisn : Pasting the error logs snippet for your reference. The attach of PVC to the pod fails with RBAC error. `volumeattachments.storage.k8s.io` has only `get` access in the clusterrole `system:node`. The workaround I followed is to add more permissions like `create,delete,update` for `volumeattachments.storage.k8s.io` to get it working.

```
Feb  6 23:36:01 kube-bol9ch0d097r490nmcgg-ocs430publi-pool645-0000059a kubelet.service: E0206 23:36:01.154844   31240 nestedpendingoperations.go:270] Operation for "\"kubernetes.io/csi/openshift-storage.rbd.csi.ceph.com^0001-0011-openshift-storage-0000000000000001-83a8cc4a-496b-11ea-8688-0a4ef85f1c8a\"" failed. No retries permitted until 2020-02-06 23:36:01.654805271 -0600 CST m=+1170801.186458985 (durationBeforeRetry 500ms). Error: "AttachVolume.Attach failed for volume \"pvc-eb8fb041-1c9e-41fe-a245-3c43c3e8e136\" (UniqueName: \"kubernetes.io/csi/openshift-storage.rbd.csi.ceph.com^0001-0011-openshift-storage-0000000000000001-83a8cc4a-496b-11ea-8688-0a4ef85f1c8a\") from node \"10.94.197.234\" : kubernetes.io/csi: attacher.Attach failed: volumeattachments.storage.k8s.io is forbidden: User \"system:node:10.94.197.234\" cannot create resource \"volumeattachments\" in API group \"storage.k8s.io\" at the cluster scope: can only get individual resources of this type"
```

The purpose of this issue is today `volumeattachments.storage.k8s.io` are given `get` access and this may be set during OCS deployment I guess. This has to be updated with more permissions.

**Comment by travisn** [comment]
I still don't understand the repro. There are no reported issues of volumes failing to attach in upstream CSI or Rook that I'm aware of. Here are some questions:
1. What version of Rook, CSI, and OCS?
1. What type of PVC? What is the storage class? RBD?

@Madhu-1 Do you see what the issue could be?

**Comment by akgunjal** [comment]
@travisn : This issue is seen in OCS 4.2 latest version. So the Rook and CSI versions are associated with the OCS version I guess. This is with storage class RBD `ocs-storagecluster-ceph-rbd`

**Comment by jarrpa** [comment]
This may be a configuration issue with your OCP installation, as no one else has reported this issue. Could you check to see if you have some automation to modify the default RBAC rules OCP ships with?

**Comment by travisn** [comment]
@Madhu-1 Do we need to add `create` permission on the CSI [service accounts](https://github.com/rook/rook/blob/master/cluster/examples/kubernetes/ceph/common.yaml#L1455-L1457)? Though I don't understand why this issue has not been seen outside ibm cloud.

**Comment by Madhu-1** [comment]
@travisn we don't need create permission see RBAC in external-attacher https://github.com/kubernetes-csi/external-attacher/blob/97411fa72c72174f9b7b7a0159cb0b840f95dd34/deploy/kubernetes/rbac.yaml#L33 @akgunjal please provide the logs from the attacher container from provisioner pod for more analysis why it's failing.

**Comment by akgunjal** [comment]
@Madhu-1 @travisn @jarrpa : I did some more debugging on this issue and found that the root cause here is on our nodes, the kubelet has `enable-controller-attach-detach` set to false. Due to this the kubelet does the attach and hence I had to give the permissions to `system:node` clusterrole. I set it to `true` and then things worked fine. So I will work with our internal teams to set this to true on cluster creation. Closing this issue as OCS don't need any actions here. Thanks.
